### PR TITLE
Chore: make timeseries the default panel in 8+

### PR DESCRIPTION
**Pull request by ryantxu**
In Grafana 8, we aim to replace the flot panel with the new uPlot based one.  We should make it the default ASAP ;)

image

After we do more migrations work and better testing, we should aim to *deprecate* the graph panel in 7.5.

This PR also makes `barchart` beta and fiddles the presentation order

### CODE REVIEW

**Comment by leeoniya** [comment]
until we have better parity with the graph panel (#30564: stacking, thresholds, annotation editing), we should at least warn people about stuff that is still missing so they don't end up setting up a lot of stuff and then realizing there's no reverse-migration path and have to redo their whole config in a blinding rage.

this plays nicely into the whole "recommend the appropriate viz/panel" on our roadmap.

**Comment by ryantxu** [comment]
I'm torn -- I think the sooner people use the new goodness the better.  Then they don't have to risk migrations later.  but yes there are a couple things that are not at parity yet.  

Maybe adding a banner like this:
image

And lets aim to remove it before 7.5 release?

File: packages/grafana-data/src/panel/PanelPlugin.ts [hunk 1]
```diff
@@ -106,6 +106,7 @@ export class PanelPlugin<
   onPanelMigration?: PanelMigrationHandler<TOptions>;
   onPanelTypeChanged?: PanelTypeChangedHandler<TOptions>;
   noPadding?: boolean;
+  editBanner?: ComponentType | null;
```
Comment (ryantxu) [inline packages/grafana-data/src/panel/PanelPlugin.ts:109]: or maybe better to hardcode for `timeseries' and 'graph' migration issues

**Comment by torkelo** [comment]
The biggest potential blocker for doing this is alerting

File: public/app/plugins/panel/timeseries/EditBanner.tsx [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+      `}
+      persistenceId="timeseries-parity-azx"
+      url={'https://github.com/grafana/grafana/issues/30564'}
+      urlTitle="See open github issue"
+    >
+      <p>
+        The new time series panel is now the default panel in grafana. For most things, this is will be the best choice.
+        However a few features exist in the graph panel that are not yet implemented in this panel. Speciffically:
+        <ul>
+          <li>Annotaiton editor</li>
```
Comment (hugohaggmark) [inline public/app/plugins/panel/timeseries/EditBanner.tsx:20]: nit: small typos
```suggestion
        However, a few features exist in the graph panel that are not yet implemented in this panel. Specifically:
        <ul>
          <li>Annotation editor</li>
```

**Review by hugohaggmark — COMMENTED** [review]
Found some typos

**Comment by dprokop** [comment]
@ryantxu moving to 8.0. I was thinking about the same thing, but there is still few things we need to address before making this one a default one.

**Comment by osg-grafana** [comment]
Deferring to @achatterjee-grafana on any doc changes for this.

**Comment by oddlittlebird** [comment]
Will the old Graph panel still be an option?

**Comment by ryantxu** [comment]
> Will the old Graph panel still be an option?

For sure existing graph panels will continue to work with flot in 8.  Ideally we can remove the ability to create new flot panels in 8 like we did with the single-stat >> stat migration.  BUT there is a long list of stuff that needs done before we can consider that!  See:  https://github.com/grafana/grafana/issues/30564

**Comment by ryantxu** [comment]
closing in favor of #33639

**Pull request closed without merging**